### PR TITLE
Update to Pyramid 1.8

### LINKED ISSUE #1008: Port Kinto for Pyramid 1.8
Pyramid 1.8 is breaking a number of things:

- BasicAuth authn policy
- pserve http_port config.

Right now it is still an alpha release but it gets installed on time to time.

### PR DESCRIPTION
**Pull request by Natim**
Fixes #1008 
Revamp #1016 

- [x] Add changelog

### CODE REVIEW

**Comment by Natim** [comment]
The failing tests are because of https://github.com/Pylons/pyramid/issues/2958

**Comment by Natim** [comment]
Note: The next release of Pyramid should not alter the incoming settings.

File: docs/core/resource.rst [hunk 1]
```diff
@@ -198,7 +198,7 @@ As an example, let's build a code that will copy a collection into another:
     from pyramid import Configurator
 
 
-    config = Configurator(settings=DEFAULT_SETTINGS)
+    config = Configurator(settings={**DEFAULT_SETTINGS})
```
Comment (gabisurita) [inline docs/core/resource.rst]: Is a shallow copy here enough? Maybe it's a good idea to note with XXX that this is a bug on pyramid.

Comment (Natim) [inline docs/core/resource.rst]: Our default_settings aren't nested so yes I think it is enough.
Maybe we don't need to update the documentation about it.
The bug in Pyramid will probably be merged on Sunday but at first they said it was a feature and not a bug...

**Comment by glasserc** [comment]
So wait, finally we're not doing a shallow copy or anything?

**Comment by Natim** [comment]
> So wait, finally we're not doing a shallow copy or anything?

We don't have to because the last release of Pyramid 1.8 has my patch with a fix there.

**Pull request merged**